File: RedSheet.DbEntities/Models/RolePermission.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("RolePermissions", Schema = "dbo")]
	[Serializable]
	public partial class RolePermission  
	{
 
	[Required]
	public bool CanAdd { get; set; } 
 
	[Required]
	public bool CanDelete { get; set; } 
 
	[Required]
	public bool CanEdit { get; set; } 
 
	[Required]
	public bool CanView { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int RolePermissionId { get; set; } 
 
	[RelationshipTableAttribue("ApplicationModules","dbo")]
	//Relationships
	public int ApplicationModuleId { get; set; } 
 
	[ForeignKey("ApplicationModuleId")]
	public ApplicationModule ApplicationModule { get; set; } 
 
	[RelationshipTableAttribue("Roles","dbo")]
	//Relationships
	public int RoleId { get; set; } 
 
	[ForeignKey("RoleId")]
	public Role Role { get; set; } 

	public RolePermission()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/role-permission.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ApplicationModule, Role,  } from './'
export class RolePermission {
    constructor(rolePermission?: RolePermission )  {
        let properties = [ "canAdd", "canDelete", "canEdit", "canView", "rolePermissionId", "applicationModuleId", "roleId",];
        for (let property of properties)
            if (rolePermission && rolePermission[property])
                this[property] = rolePermission[property];
    }
 
	canAdd : boolean = false ;
 
	canDelete : boolean = false ;
 
	canEdit : boolean = false ;
 
	canView : boolean = false ;
 
	rolePermissionId : number =   0 ;
 
    @range(0,2147483647)
	applicationModuleId : number =   undefined;
	applicationModule : ApplicationModule  ;
 
    @range(0,2147483647)
	roleId : number =   undefined;
	role : Role  ;


}
